nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/coldstorage/MoveToColdStorageTest.java

```diff
@@ -42,8 +50,42 @@
     @Inject
     protected CoreSession session;
 
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Test
+    public void shouldFailWhenUserDoNotHaveWriteColdStoragePermission() throws OperationException, IOException {
+        DocumentModel documentModel = createFileDocument(session, true);
```

Comment:
I am wondering if we can keep the methods names shorter as possible -> 
`shouldFailWithoutRightPermissions` we don't need to explicitly put the permission in the name of the method, which will avoid renaming the method if the permission changes its names in the future

Reply:
We can, but I would say it may not be a consensual approach.
Some schools of thought advocate that unit test names should be clear and map specific requirements/acceptance criteria and, in this case, we can end with long names. Because, at the end of the day, who reads the test name should understand what is being tested.

If the permissions change in future, the requirements changed and we probably want to change the tests too 🙂 

Anyway, I can try to follow other approach for consistency sake. What is our convention?

Reply:
@nmpcunha, unfortunately, for now, we don't have such conventions.
We plan to provide them in a near future.

So, I guess it's the responsibility of the writer to choose something that is at least consistent with the rest of the test/package/Maven module.
In this case, I would at least replace `WhenUserDoesNotHave` by the suggested `Without`.
We really want to keep the test names as short as possible for readability in the test logs, the CI unit test reports, etc.
As for the permission name, that's your choice, though personally I like @RSalem07 's approach, simply because it's shorter.

Reply:
Thanks for your input! 👍 
I've reworked the tests according to your suggestions. Hopefully we can work together on those conventions 🙂